lib/krane/runner_task.rb

```diff
@@ -98,11 +104,11 @@ def create_pod(pod)
       raise FatalDeploymentError, msg
     end
 
-    def build_pod(template_name, command, args, env_vars, verify_result)
+    def build_pod(template_name, container_overrides, verify_result)
       task_template = get_template(template_name)
       @logger.info("Using template '#{template_name}'")
       pod_template = build_pod_definition(task_template)
-      set_container_overrides!(pod_template, command, args, env_vars)
+      set_container_overrides!(pod_template, container_overrides)
```

Comment:
Should we move this entire function into the ContainerOverrides class and make this line something like
`pod_template = container_overrides.run!(pod_template)`

Reply:
I not entirely. If we do that we create a dependency that `container_overrides` can only by run by `task_runner`

Since we have a validation before overriding the container.

```ruby
def set_container_overrides!(pod_definition, container_overrides)
  container = pod_definition.spec.containers.find { |cont| cont.name == 'task-runner' }
  if container.nil?
    message = "Pod spec does not contain a template container called 'task-runner'"
    @logger.summary.add_paragraph(message)
    raise TaskConfigurationError, message
  end

  container_overrides.run!(container)
end
```

I think is better if we leave the validation inside `task_runner`

Reply:
I can modify the method `build_pod` create a new method called `extract_task_runner_container` which only takes care of the validation from above.

And finally, we can simply run `container_overrides.run!(container)`

Reply:
I like your proposal, though it isn't a blocker for me